src/main/host/process.c

```diff
@@ -833,6 +833,28 @@ PluginPhysicalPtr process_getPhysicalAddress(Process* proc, PluginVirtualPtr vPt
     return (PluginPhysicalPtr){.val = low | high};
 }
 
+int process_readPtr(Process* proc, Thread* thread, void* dst, PluginVirtualPtr src, size_t n) {
+    MAGIC_ASSERT(proc);
+    if (proc->memoryManager) {
+        const void* mapped = memorymanager_getReadablePtr(proc->memoryManager, thread, src, n);
+        memcpy(dst, mapped, n);
+        return 0;
```

Comment:
I think the main APIs to use in the syscallhandlers going forward will be worker_*, and we can build safe(r) Rust interfaces on top of those. The "typed" memorymanager interfaces that are currently sketched out but unused will probably end up being removed, though we can reuse some of the API ideas for the new APIs.

In particular `readPtr` and `writePtr` makes less sense for the `MemoryManager`, since its main purpose is to remap pointers - it can let the caller deal with copies if they're needed.